vectorizing/solvers/binary/bitmap.py

```diff
@@ -29,34 +29,47 @@ def alpha_blend(img_arr):
 
     return cv2.merge((r, g, b)).astype(np.uint8)
 
-# Creates bitmap to be traced by potrace
-def make_bitmap(img_arr):
-    shape = img_arr.shape
-    channel_count = shape[-1]
-    pixel_count = shape[0] * shape[1]
-    inverted_arr = invert(img_arr)
+
+# Compute threshold
+def threshold(img_arr):
+    channel_count = img_arr.shape[-1]
 
     if channel_count == 4:
         img_arr = alpha_blend(img_arr)
-        inverted_arr = alpha_blend(inverted_arr)
-    
+
     img_arr = cv2.cvtColor(img_arr, cv2.COLOR_RGB2GRAY)
-    inverted_arr = cv2.cvtColor(inverted_arr, cv2.COLOR_RGB2GRAY)
-    
-    bitmap = threshold(img_arr)
-    inverted_bitmap = threshold(inverted_arr)
+    thresholded = cv2.threshold(img_arr, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)[
+        1
+    ].astype(np.uint8)
 
+    # Dark is foreground
+    return np.where(thresholded >= 128, 0, 1)
+
+
+# Creates bitmap to be traced by potrace
+# This deals with a very common edge case: when an image has a transparent
+# background and a white foreground, the alpha blend strips all information, so in
+# that case we choose the inverted version
+def compute_bitmap(img_arr, foreground_area_threshold=0.01):
+    # compute bitmap
+    bitmap = threshold(img_arr)
     foreground_area = np.sum(bitmap)
+    pixel_count = img_arr.shape[0] * img_arr.shape[1]
 
-    if foreground_area == 0:
-        # If the non-inverted foreground has an area of zero, 
-        # return a black rectangle
-        # i.e Fully transparent image
-        return np.ones(bitmap.shape)
+    # If the total area of the foreground is less than the defined threshold, check if
+    # it's better to use the inverted bitmap instead
+    chosen_image = bitmap
+    if foreground_area / pixel_count < foreground_area_threshold:
```

Comment:
`pixel_count` could be zero, but I think it's fine, we need to add some image validation before to make sure this is not possible (will come with tests), but we should keep it in mind